pkg/kubelet/kubelet.go

```diff
@@ -920,10 +926,20 @@ func (kl *Kubelet) listPodsFromDisk() ([]types.UID, error) {
 // Starts garbage collection threads.
 func (kl *Kubelet) StartGarbageCollection() {
 	go wait.Until(func() {
-		if err := kl.containerGC.GarbageCollect(kl.sourcesReady.AllReady()); err != nil {
-			glog.Errorf("Container garbage collection failed: %v", err)
+		kl.gcController.containerGCTrigger = make(chan struct{}, 1)
```

Comment:
suggestion: move gcController to a separate file and returns a channel at initialization